kore/src/Kore/Step/Function/Evaluator.hs

```diff
@@ -249,41 +249,42 @@ maybeEvaluatePattern
                 Profile.axiomEvaluation identifier
                 $ evaluator termLike sideConditions
             flattened <- case result of
-                AttemptedAxiom.NotApplicable -> do
+                AttemptedAxiom.NotApplicable ->
                     DebugAxiomEvaluation.notEvaluated
```

Comment:
Why does this need to be wrapped around return? Nothing will be logged during this time, so this can just be a log entry (instead of a context).